### ISSUE TITLE
Crash in ProcessReads.cpp:2148 while trying to make a genomebam.

### ISSUE THREAD

**Issue by detrout** [pachterlab/kallisto#289]
Hi,

Using kallisto 0.46.2 compiled from source with this command line:
```
kallisto quant -t 6 \
  -i mm10-M21_minimal-male/mm10-M21_minimal-male-kallisto-0.46.2.idx \
  -o 19906_A5 \
  --single \
  --fragment-length=1759.25 \
  --sd=1103.91175 \
  --genomebam \
  --gtf mm10-M21_minimal-male/mm10-M21_minimal-male.gtf  \
  19906_A5_GGACTCCT-CTCTCTAT_L001_R1_001.fastq.gz \
  19906_A5_GGACTCCT-CTCTCTAT_L002_R1_001.fastq.gz
```

I get a crash here:

(gdb) bt
```
#0  AlnProcessor::processBufferGenome (this=0x55555730b418) at /woldlab/loxcyc/home/diane/proj/kallisto/src/ProcessReads.cpp:2148
#1  0x00005555556fb47d in AlnProcessor::operator() (this=0x55555730b418) at /woldlab/loxcyc/home/diane/proj/kallisto/src/ProcessReads.cpp:1679
#2  0x000055555570a4dd in std::__invoke_impl<void, AlnProcessor> (__f=...) at /usr/include/c++/8/bits/invoke.h:60
#3  0x0000555555706238 in std::__invoke<AlnProcessor> (__fn=...) at /usr/include/c++/8/bits/invoke.h:95
#4  0x000055555571c12c in std::thread::_Invoker<std::tuple<AlnProcessor> >::_M_invoke<0ul> (this=0x55555730b418) at /usr/include/c++/8/thread:244
#5  0x000055555571c0ce in std::thread::_Invoker<std::tuple<AlnProcessor> >::operator() (this=0x55555730b418) at /usr/include/c++/8/thread:253
#6  0x000055555571c072 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<AlnProcessor> > >::_M_run (this=0x55555730b410)
    at /usr/include/c++/8/thread:196
#7  0x00007ffff7c99b2f in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#8  0x00007ffff7f91fa3 in start_thread (arg=<optimized out>) at pthread_create.c:486
#9  0x00007ffff79774cf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
(gdb) 
```

The problem appears to be at this block of code:
```
    if (mp.opt.busOptions.seq.front().fileno == 1) {
      std::swap(si1,si2);
      std::swap(rlen1,rlen2);
    }
```
as 
```
(gdb) p mp.opt.busOptions.seq
$9 = std::vector of length 0, capacity 0
```
since seq is a null value, I suppose it could be fixed by doing this instead:
```
if (mp.opt.busOptions.seq && mp.opt.busOptions.seq.front().fileno == 1) {
```
Though I'm not sure if busOptions.seq is supposed to be set by some command line argument.

**Comment by detrout** [comment]
Er  seq is not a pointer, its a vector, so this worked.

```
index ecaca8a..889ee5f 100644
--- a/src/ProcessReads.cpp
+++ b/src/ProcessReads.cpp
@@ -2145,7 +2145,7 @@ void AlnProcessor::processBufferGenome() {
       rlen2 = seqs[si2].second;
     }
     
-    if (mp.opt.busOptions.seq.front().fileno == 1) {
+    if (!mp.opt.busOptions.seq.empty() && mp.opt.busOptions.seq.front().fileno == 1) {
       std::swap(si1,si2);
       std::swap(rlen1,rlen2);
     }
```

**Comment by Yenaled** [comment]
Fixed in 0.48.0
